rust/azure_iot_operations_connector/src/filemount/observability_artifacts.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Types for extracting Observability configurations from an AIO deployment.
+//! Note that this module is a stopgap implementation and will be replaced in
+//! the future by a unified approach to observability endpoints.
+
+use std::path::PathBuf;
+
+/// Values extracted from the observability artifacts in an AIO deployment.
+pub struct ObservabilityArtifacts {
+    /// OTEL grpc/grpcs metric endpoint.
+    pub grpc_metric_endpoint: Option<String>,
+    /// OTEL grpc/grpcs log endpoint.
+    pub grpc_log_endpoint: Option<String>,
+    /// OTEL grpc/grpcs trace endpoint.
+    pub grpc_trace_endpoint: Option<String>,
+    /// Path to the directory containing trust bundle for 1P grpc metric collector.
+    pub grpc_metric_collector_1p_ca_mount: Option<PathBuf>,
+    /// Path to the directory containing trust bundle for 1P grpc log collector.
+    pub grpc_log_collector_1p_ca_mount: Option<PathBuf>,
+
+    /// OTEL http/https metric endpoint.
+    pub http_metric_endpoint: Option<String>,
+    /// OTEL http/https log endpoint.
+    pub http_log_endpoint: Option<String>,
+    /// OTEL http/https trace endpoint.
+    pub http_trace_endpoint: Option<String>,
+
+    /// OTEL 3P metric endpoint.
+    pub metric_endpoint_3p: Option<String>,
+    /// OTEL 3P metric export interval.
+    pub metric_export_interval_3p: Option<i32>,
+}
+
+impl ObservabilityArtifacts {
+    /// Create an `ObservabilityArtifacts` instance from the environment variables in an AIO deployment.
+    pub fn new_from_deployment() -> ObservabilityArtifacts {
+        let grpc_metric_endpoint = std::env::var("OLTP_GRPC_METRIC_ENDPOINT").ok();
+        let grpc_log_endpoint = std::env::var("OLTP_GRPC_LOG_ENDPOINT").ok();
+        let grpc_trace_endpoint = std::env::var("OLTP_GRPC_TRACE_ENDPOINT").ok();
+
+        let grpc_metric_collector_1p_ca_mount =
+            std::env::var("FIRST_PARTY_OTLP_GRPC_METRICS_COLLECTOR_CA_PATH")
+                .ok()
+                .map(PathBuf::from);
+        let grpc_log_collector_1p_ca_mount =
+            std::env::var("FIRST_PARTY_OTLP_GRPC_LOG_COLLECTOR_CA_PATH")
+                .ok()
+                .map(PathBuf::from);
+
+        let http_metric_endpoint = std::env::var("OLTP_HTTP_METRIC_ENDPOINT").ok();
+        let http_log_endpoint = std::env::var("OLTP_HTTP_LOG_ENDPOINT").ok();
+        let http_trace_endpoint = std::env::var("OLTP_HTTP_TRACE_ENDPOINT").ok();
+
+        let metric_endpoint_3p = std::env::var("OTLP_METRIC_ENDPOINT_3P").ok();
+        let metric_export_interval_3p = std::env::var("OTLP_METRIC_EXPORT_INTERVAL_3P")
+            .map(|v| v.parse::<i32>())
+            .ok()
+            .transpose()
+            .ok()
```

Comment:
doesn't this just go back and forth?
```suggestion
            .ok()
```